uiowa/uiowa
# Sitenow Search module

Resolves #3373 

Ability to search all text fields for a given string or regular expression, and returns those field types/content and the node on which they appear. Currently doesn't search revisions, so only checks current 'published' text fields.





# How to test



- Sync a site
- `drush @site.local uli /admin/config`
- Go to the Sitenow Find Text page
- Try searching for some stuff, see if it works!
  - Node titles and fields directly attached to nodes should be searchable.
  - Fields attached to blocks should be searchable.
  - Fields attached to paragraphs (attached to blocks) should be searchable.
    - The focus was on v3, but paragraph content on v2 should be searchable as well.
  - Content (nodes and blocks) included as feature splits (such as AoS) should be searchable as well as standard platform content.
  - Menu items/links are not searchable.
  - Reusable blocks (such as in the footer) are not searchable.
  - Contents pulled in from other sources (directory profiles, aggregator content) are not searchable.
  - Note: Some default content may not show up such as Home page blocks--this is expected behavior.
- Masquerade as a webmaster and check that the page is available and you are able to perform content searches
- Sync some sites with various splits and check that they sync correctly, the module is available, and webmaster has permissions

Scanner offers some additional options, such as matching on word boundaries, limiting to published pages only, etc. (https://www.drupal.org/project/scanner) Some of these may be useful, some not, and some may be more useful if this was opened up to content editors, but not necessary for more experienced users who may be familiar with LIKE and regex options.

Also note for a possible followup: currently doesn't search menu items, which for things like #3996, is an extra consideration.
